src/lib/comms/include/sol-bluetooth.h

```diff
@@ -0,0 +1,327 @@
+ *
+ * @param conn The reference to a connection.
+ *
+ * @return 0 on success, -errno otherwise.
+ */
+int sol_bt_disconnect(struct sol_bt_conn *conn);
+
+/**
+ * @brief Represents a Bluetooth usage session
+ *
+ * Because Bluetooth usage may increase the power comsuption, there's
+ * a need to keep track of what is using Bluetooth in the system, and
+ * keep Bluetooth turned off if it's not used.
+ */
+struct sol_bt_session;
+
+/**
+ * @brief Enables the local Bluetooth controller.
+ *
+ * Before using any other functionality from this module, this function
+ * should be called.
+ *
+ * @param enabled Function to be called when the controller changes state.
+ * @param user_data User data to be provided to the enabled() callback.
+ *
+ * @return a reference to a session, used for returning the system to its
+ * previous state, @see sol_bt_disable().
+ */
+struct sol_bt_session *sol_bt_enable(
+    void (*enabled)(void *data, bool powered),
+    const void *user_data);
+
+/**
+ * @brief Disables a session, returning the controller to its previous state.
+ *
+ * @param session Reference to a session.
+ *
+ * @return 0 on success, -errno otherwise.
+ */
+int sol_bt_disable(struct sol_bt_session *session);
+
+/**
+ * @brief Represents a information about a remote device.
+ */
+struct sol_bt_device_info {
+    /**
+     * @brief Network address of the remote device.
+     */
+    struct sol_network_link_addr addr;
+    /**
+     * @brief Vector of service UUIDs discovered, may be empty.
+     */
+    struct sol_vector uuids;
+    /**
+     * @brief Friendly name of the device.
+     */
+    char *name;
+    /**
+     * @brief Received signal strength, measured in dBm.
+     */
+    int16_t rssi;
+    /**
+     * @brief Whether the device is paired.
+     */
+    bool paired;
+    /**
+     * @brief Whether the device is connected.
+     */
+    bool connected;
+    /**
+     * @brief Whether the devices is currently in range.
+     */
+    bool in_range;
+};
+
+/**
+ * @brief Over which transport should a scan be performed.
+ */
+enum sol_bt_transport {
+    /** @brief Discover devices over All transports */
+    SOL_BT_TRANSPORT_ALL,
+    /** @brief Discover devices over the Bluetooth Low Energy transport */
+    SOL_BT_TRANSPORT_LE,
+    /** @brief Discover devices over the Bluetooth Basic Rate transport */
+    SOL_BT_TRANSPORT_BREDR,
+};
+
+/**
+ * @brief Converts a transport to a string, @see sol_bt_transport.
+ *
+ * @param transport A transport.
+ *
+ * @return a pointer to a string representing the transport on success,
+ *         NULL otherwise.
+ */
+const char *sol_bt_transport_to_str(enum sol_bt_transport transport);
+
+/**
+ * @brief Converts string to a transport, @see sol_bt_transport.
+ *
+ * @param str A with the transport representation.
+ *
+ * @return a pointer to a string representing the transport on success,
+ *         NULL otherwise.
+ */
+enum sol_bt_transport sol_bt_transport_from_str(const char *str);
```

Comment:
That seems to be the convention when writing code that will convert to/from enums.

As this kind of conversion will be mostly used by language bindings, this is easier it seems.